front/components/Dashboard/Dashboard.tsx

```diff
@@ -0,0 +1,149 @@
+import * as React from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import { Redirect } from 'react-router';
+import styled from 'styled-components';
+
+import { API_ENDPOINT } from '../../settings';
+import { Video, videoState } from '../../types/Video';
+import { Maybe } from '../../utils/types';
+import { ROUTE as ERROR_ROUTE } from '../ErrorComponent/ErrorComponent';
+import { H6, IframeHeading } from '../Headings/Headings';
+import { UploadStatusList } from '../UploadStatusList/UploadStatusList';
+import { DashboardButtons } from './DashboardButtons';
+import { DashboardHelptext } from './DashboardHelptext';
+
+const { UPLOADING } = videoState;
+
+const messages = defineMessages({
+  subtitleForVideo: {
+    defaultMessage: 'Video preparation',
+    description:
+      'Subtitle for the video part of the dashboard (right now the only part until we add subtitles).',
+    id: 'components.Dashboard.subtitleForVideo',
+  },
+  title: {
+    defaultMessage: 'Dashboard',
+    description: `Title for the dashboard, where the user can see the status of the video/audio/subtitles upload
+      and processing, will be able to manage them.`,
+    id: 'components.Dashboard.title',
+  },
+});
+
+export const ROUTE = () => '/dashboard';
+
+const IframeHeadingWithLayout = styled(IframeHeading)`
+  flex-grow: 0;
+  margin: 0;
+  text-align: center;
+`;
+
+const DashboardContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: stretch;
+  height: 100%;
+`;
+
+const DashboardInternalHeading = styled(H6)`
+  margin: 0;
+  padding: 0 1rem;
+  font-weight: 400;
+`;
+
+const DashboardInnerContainer = styled.div`
+  flex-grow: 1;
+  display: flex;
```

Comment:
I usually define `display: xxx` first and then other properties such as flex-related ones in this case. By doing so, we know what kind of object we are dealing with before going further.

Reply:
That's what I do too but here `flex-grow` does not interact with this `display: flex`. It makes the element grow in the context of the parent `display:flex` element, and then sets `display: flex` to layout the children